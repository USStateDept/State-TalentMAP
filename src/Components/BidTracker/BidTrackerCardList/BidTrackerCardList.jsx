import React from 'react';
import PropTypes from 'prop-types';
import { BID_RESULTS } from '../../../Constants/PropTypes';
import BidTrackerCard from '../BidTrackerCard';

const BidTrackerCardList = ({ bids, acceptBid, declineBid }) => (
  <div className="usa-grid-full">
    {
      bids.map(bid => (
        <BidTrackerCard key={bid.id} bid={bid} acceptBid={acceptBid} declineBid={declineBid} />
      ))
    }
  </div>
);

BidTrackerCardList.propTypes = {
  bids: BID_RESULTS.isRequired,
  acceptBid: PropTypes.func.isRequired,
  declineBid: PropTypes.func.isRequired,
};

export default BidTrackerCardList;
