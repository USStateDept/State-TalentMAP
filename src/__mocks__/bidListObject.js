const bidListObject = {
  count: 3,
  next: null,
  previous: null,
  results: [
    {
      id: 3,
      bidcycle: 'Demo BidCycle 2017-10-31 16:00:51.254762',
      user: 'rehmant',
      position: {
        id: 226,
        grade: '03',
        skill: 'POLITICAL AND ECONOMIC AFFAIRS (6050)',
        position_number: '10035561',
        title: 'POLITICAL/ECONOMIC OFFICER',
        create_date: '2015-08-05',
        update_date: '2017-06-08',
        bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
        post: {
          id: 199,
          location: 'Curacao',
        },
      },
      status: 'approved',
      submission_date: '2017-10-31',
      update_date: '2017-10-31',
      reviewer: {
        username: 'woodwardw',
        first_name: 'Wendy',
        last_name: 'Woodward',
        email: 'woodwardw@state.gov',
        phone_number: '555-555-5555',
        is_cdo: false,
      },
    },
    {
      id: 1,
      bidcycle: 'Demo BidCycle 2017-10-31 16:00:51.254762',
      user: 'rehmant',
      position: {
        id: 1,
        grade: '05',
        skill: 'OFFICE MANAGEMENT (9017)',
        position_number: '00003026',
        title: 'OMS (COM)',
        create_date: '2006-09-20',
        update_date: '2017-06-08',
        bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
        post: {
          id: 235,
          location: 'Freetown, Sierra Leone',
        },
      },
      status: 'closed',
      submission_date: '2017-10-31',
      update_date: '2017-11-01',
      reviewer: {
        username: 'woodwardw',
        first_name: 'Wendy',
        last_name: 'Woodward',
        email: 'woodwardw@state.gov',
        phone_number: '555-555-5555',
        is_cdo: false,
      },
    },
    {
      id: 2,
      bidcycle: 'Demo BidCycle 2017-10-31 16:00:51.254762',
      user: 'rehmant',
      position: {
        id: 84,
        grade: '04',
        skill: 'SECURITY (2501)',
        position_number: '56001600',
        title: 'SPECIAL AGENT',
        create_date: '2006-09-20',
        update_date: '2017-06-08',
        bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
        post: {
          id: 195,
          location: 'The Hague, Netherlands',
        },
      },
      status: 'closed',
      submission_date: null,
      update_date: '2017-11-01',
      reviewer: {
        username: 'woodwardw',
        first_name: 'Wendy',
        last_name: 'Woodward',
        email: 'woodwardw@state.gov',
        phone_number: '555-555-5555',
        is_cdo: false,
      },
    },
  ],
};

export default bidListObject;
