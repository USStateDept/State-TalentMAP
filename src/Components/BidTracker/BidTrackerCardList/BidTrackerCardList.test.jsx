import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import BidTrackerCardList from './BidTrackerCardList';
import bidListObject from '../../../__mocks__/bidListObject';

describe('BidTrackerCardListComponent', () => {
  const bids = bidListObject.results;

  const props = {
    bids,
    acceptBid: () => {},
    declineBid: () => {},
  };

  it('is defined', () => {
    const wrapper = shallow(
      <BidTrackerCardList {...props} />,
    );
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <BidTrackerCardList {...props} />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
