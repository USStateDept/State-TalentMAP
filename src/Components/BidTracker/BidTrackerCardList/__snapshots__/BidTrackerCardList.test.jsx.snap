// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BidTrackerCardListComponent matches snapshot 1`] = `
<div
  className="usa-grid-full"
>
  <BidTrackerCardContainer
    acceptBid={[Function]}
    bid={
      Object {
        "approved_date": "2017-12-20",
        "bidcycle": "Demo BidCycle 2017-10-31 16:00:51.254762",
        "closed_date": "2017-12-20",
        "create_date": "2017-12-20",
        "declined_date": "2017-12-20",
        "draft_date": "2017-12-20",
        "handshake_accepted_date": "2017-12-20",
        "handshake_declined_date": "2017-12-20",
        "handshake_offered_date": "2017-12-20",
        "id": 1,
        "in_panel_date": "2017-12-20",
        "is_priority": true,
        "position": Object {
          "bureau": "(AF) BUREAU OF AFRICAN AFFAIRS",
          "create_date": "2006-09-20",
          "grade": "05",
          "id": 1,
          "position_number": "00003026",
          "post": Object {
            "id": 235,
            "location": "Freetown, Sierra Leone",
          },
          "skill": "OFFICE MANAGEMENT (9017)",
          "title": "OMS (COM)",
          "update_date": "2017-06-08",
        },
        "reviewer": Object {
          "email": "woodwardw@state.gov",
          "first_name": "Wendy",
          "is_cdo": false,
          "last_name": "Woodward",
          "phone_number": "555-555-5555",
          "username": "woodwardw",
        },
        "scheduled_panel_date": "2017-12-20",
        "status": "closed",
        "submitted_date": "2017-12-20",
        "update_date": "2017-12-20",
        "user": "rehmant",
      }
    }
    declineBid={[Function]}
    deleteBid={[Function]}
    isPriority={false}
    priorityExists={true}
    submitBid={[Function]}
    userProfile={
      Object {
        "bid_statistics": Array [
          Object {
            "draft": 1,
            "submitted": 1,
          },
        ],
        "cdo": Object {
          "email": "shadtrachl@state.gov",
          "first_name": "Leah",
          "last_name": "Shadtrach",
        },
        "display_name": "John",
        "favorite_positions": Array [
          Object {
            "id": 1,
            "representation": "[00003026] OMS (COM) (Freetown, Sierra Leone)",
          },
          Object {
            "id": 4,
            "representation": "[00180000] OMS (DCM) (Addis Ababa, Ethiopia)",
          },
        ],
        "grade": "03",
        "id": 1,
        "initials": "JD",
        "is_cdo": false,
        "skills": Array [
          Object {
            "code": "6218",
            "cone": null,
            "description": "CONSTRUCTION ENGINEERING",
            "id": 63,
          },
          Object {
            "code": "6211",
            "cone": null,
            "description": "COMPUTER SCIENCE",
            "id": 40,
          },
        ],
        "user": Object {
          "email": "doej@state.gov",
          "first_name": "John",
          "last_name": "Doe",
          "username": "jdoe",
        },
      }
    }
  />
  <BidTrackerCardContainer
    acceptBid={[Function]}
    bid={
      Object {
        "approved_date": "2017-12-20",
        "bidcycle": "Demo BidCycle 2017-10-31 16:00:51.254762",
        "closed_date": "2017-12-20",
        "create_date": "2017-12-20",
        "declined_date": "2017-12-20",
        "draft_date": "2017-12-20",
        "handshake_accepted_date": "2017-12-20",
        "handshake_declined_date": "2017-12-20",
        "handshake_offered_date": "2017-12-20",
        "id": 2,
        "in_panel_date": "2017-12-20",
        "is_priority": false,
        "position": Object {
          "bureau": "(AF) BUREAU OF AFRICAN AFFAIRS",
          "create_date": "2006-09-20",
          "grade": "04",
          "id": 84,
          "position_number": "56001600",
          "post": Object {
            "id": 195,
            "location": "The Hague, Netherlands",
          },
          "skill": "SECURITY (2501)",
          "title": "SPECIAL AGENT",
          "update_date": "2017-06-08",
        },
        "reviewer": Object {
          "email": "woodwardw@state.gov",
          "first_name": "Wendy",
          "is_cdo": false,
          "last_name": "Woodward",
          "phone_number": "555-555-5555",
          "username": "woodwardw",
        },
        "scheduled_panel_date": "2017-12-20",
        "status": "closed",
        "submitted_date": "2017-12-20",
        "update_date": "2017-12-20",
        "user": "rehmant",
      }
    }
    declineBid={[Function]}
    deleteBid={[Function]}
    isPriority={false}
    priorityExists={true}
    submitBid={[Function]}
    userProfile={
      Object {
        "bid_statistics": Array [
          Object {
            "draft": 1,
            "submitted": 1,
          },
        ],
        "cdo": Object {
          "email": "shadtrachl@state.gov",
          "first_name": "Leah",
          "last_name": "Shadtrach",
        },
        "display_name": "John",
        "favorite_positions": Array [
          Object {
            "id": 1,
            "representation": "[00003026] OMS (COM) (Freetown, Sierra Leone)",
          },
          Object {
            "id": 4,
            "representation": "[00180000] OMS (DCM) (Addis Ababa, Ethiopia)",
          },
        ],
        "grade": "03",
        "id": 1,
        "initials": "JD",
        "is_cdo": false,
        "skills": Array [
          Object {
            "code": "6218",
            "cone": null,
            "description": "CONSTRUCTION ENGINEERING",
            "id": 63,
          },
          Object {
            "code": "6211",
            "cone": null,
            "description": "COMPUTER SCIENCE",
            "id": 40,
          },
        ],
        "user": Object {
          "email": "doej@state.gov",
          "first_name": "John",
          "last_name": "Doe",
          "username": "jdoe",
        },
      }
    }
  />
  <BidTrackerCardContainer
    acceptBid={[Function]}
    bid={
      Object {
        "approved_date": "2017-12-20",
        "bidcycle": "Demo BidCycle 2017-10-31 16:00:51.254762",
        "closed_date": "2017-12-20",
        "create_date": "2017-12-20",
        "declined_date": "2017-12-20",
        "draft_date": "2017-12-20",
        "handshake_accepted_date": "2017-12-20",
        "handshake_declined_date": "2017-12-20",
        "handshake_offered_date": "2017-12-20",
        "id": 3,
        "in_panel_date": "2017-12-20",
        "is_priority": false,
        "position": Object {
          "bureau": "(AF) BUREAU OF AFRICAN AFFAIRS",
          "create_date": "2015-08-05",
          "grade": "03",
          "id": 226,
          "position_number": "10035561",
          "post": Object {
            "id": 199,
            "location": "Curacao",
          },
          "skill": "POLITICAL AND ECONOMIC AFFAIRS (6050)",
          "title": "POLITICAL/ECONOMIC OFFICER",
          "update_date": "2017-06-08",
        },
        "reviewer": Object {
          "email": "woodwardw@state.gov",
          "first_name": "Wendy",
          "is_cdo": false,
          "last_name": "Woodward",
          "phone_number": "555-555-5555",
          "username": "woodwardw",
        },
        "scheduled_panel_date": "2017-12-20",
        "status": "approved",
        "submitted_date": "2017-12-20",
        "update_date": "2017-12-20",
        "user": "rehmant",
      }
    }
    declineBid={[Function]}
    deleteBid={[Function]}
    isPriority={false}
    priorityExists={true}
    submitBid={[Function]}
    userProfile={
      Object {
        "bid_statistics": Array [
          Object {
            "draft": 1,
            "submitted": 1,
          },
        ],
        "cdo": Object {
          "email": "shadtrachl@state.gov",
          "first_name": "Leah",
          "last_name": "Shadtrach",
        },
        "display_name": "John",
        "favorite_positions": Array [
          Object {
            "id": 1,
            "representation": "[00003026] OMS (COM) (Freetown, Sierra Leone)",
          },
          Object {
            "id": 4,
            "representation": "[00180000] OMS (DCM) (Addis Ababa, Ethiopia)",
          },
        ],
        "grade": "03",
        "id": 1,
        "initials": "JD",
        "is_cdo": false,
        "skills": Array [
          Object {
            "code": "6218",
            "cone": null,
            "description": "CONSTRUCTION ENGINEERING",
            "id": 63,
          },
          Object {
            "code": "6211",
            "cone": null,
            "description": "COMPUTER SCIENCE",
            "id": 40,
          },
        ],
        "user": Object {
          "email": "doej@state.gov",
          "first_name": "John",
          "last_name": "Doe",
          "username": "jdoe",
        },
      }
    }
  />
</div>
`;
